### PR TITLE
add HierarchicalError interface

### PR DESCRIPTION
**Pull request by kovetskiy**

### CODE REVIEW

File: hierr.go [hunk 1]
```diff
@@ -80,6 +80,13 @@ type Error struct {
 	Nested interface{}
 }
 
+// HierearchicalError represents interface, which methods will be used instead
```
Comment (seletskiy) [inline hierr.go]: s/Hiere/Hier

File: hierr.go [hunk 2]
```diff
@@ -80,6 +80,13 @@ type Error struct {
...
+// of calling String() and Error() methods.
+type HierarchicalError interface {
+	// HierearchicalError returns hierarhical string representation.
```
Comment (seletskiy) [inline hierr.go]: Same typo here.

File: hierr.go (hunk 1 above)
Comment (kovetskiy) [inline hierr.go]: @seletskiy force pushd

**Comment by seletskiy** [comment]
@kovetskiy: Great work!

**Pull request merged**